lms/config/resources.py

```diff
@@ -134,6 +139,14 @@ def h_provider_unique_id(self):
         """
         return self._get_param("user_id")
 
+    @property
+    def h_pubid(self):
+        return self._h_pubid
+
+    @h_pubid.setter
+    def h_pubid(self, new_value):
+        self._h_pubid = new_value
```

Comment:
The part of the code that sends the group `PUT` or `PATCH request to h receives the group's `pubid` from h. But the `pubid` is needed in other parts of the code, elsewhere in the request processing cycle. So I made it available on the `context` object.

Reply:
Given that there is a publicly-available setter here, I've lost the thread on why the private attribute (`_h_pubid`) is needed (versus just a public property)?

Reply:
You mean why `LTILaunch.h_pubid` is not just a simple instance attribute, instead of being a `@property` with a getter and setter? It's because it makes autospeccing work correctly

Reply:
Also, later on this might be private, it occurs to me.